service/matching/incoming_nexus_service_manager.go

```diff
@@ -286,9 +303,11 @@ func (m *incomingServiceManager) loadServices(ctx context.Context) error {
 			m.servicesByName[entry.Service.Name] = entry
 		}
 
-		finishedPaging = pageToken == nil
+		if pageToken == nil {
+			break
+		}
 	}
 
-	m.hasLoadedServices.Store(finishedPaging)
+	m.hasLoadedServices.Store(pageToken == nil)
```

Comment:
Can you explain why you store `false` if page token is not nil? Page token may be nil before the first page has been returned.
Is this meant to be an indication that `ctx.Err()` was not nil and we stopped in the middle of loading?
Seems like we shouldn't set only when pageToken is nil in before breaking out of the loop.

Reply:
Yeah, it's meant to be an indication that we encountered an error during paging and therefore did not completely load all services (e.g. timeout). In that case we would want to try reloading before serving requests from memory.

I guess that is probably only the case when there is a context error, so setting it to `ctx.Err() == nil` makes more sense.